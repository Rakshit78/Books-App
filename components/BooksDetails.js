import React from 'react';
import {
  Modal,
  View,
  Text,
  TouchableOpacity,
  Image,
  ScrollView,
  StyleSheet,
} from 'react-native';
import { BlurView } from 'expo-blur';
import AntDesign from '@expo/vector-icons/AntDesign';

function BooksDetails({ modal, setModal }) {
  return (
    <>
      <Modal transparent visible={modal !== null} animationType='slide'>
        <BlurView style={{ flex: 1 }} intensity={20}>
          <View style={styles.container}>
            <TouchableOpacity onPress={() => setModal(null)} style={styles.btn}>
              <AntDesign name='close' size={24} color='#FFF' />
            </TouchableOpacity>
            <View style={styles.container1}>
              <Image
                source={{
                  uri: modal?.volumeInfo?.imageLinks?.thumbnail,
                }}
                resizeMode='cover'
                style={styles.image}
              />
              <Text style={styles.title} numberOfLines={2}>
                {modal?.volumeInfo?.title}
              </Text>
              <Text style={styles.authors} numberOfLines={1}>
                {modal?.volumeInfo?.authors}
              </Text>
              <Text style={styles.publisher}>
                Published Date: {modal?.volumeInfo?.publishedDate}
              </Text>
              <Text style={styles.date}>
                Publisher:{' '}
                {modal?.volumeInfo?.publisher
                  ? modal?.volumeInfo?.publisher
                  : 'Not Available'}
              </Text>
            </View>
            <ScrollView style={styles.descriptionContainer}>
              <Text style={styles.descTitle}>Description</Text>
              <Text style={styles.description}>
                {modal?.volumeInfo?.description}
              </Text>
            </ScrollView>
          </View>
        </BlurView>
      </Modal>
    </>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#1A1A1A',
    alignItems: 'center',
    marginTop: '35%',
    borderTopEndRadius: 15,
    borderTopLeftRadius: 15,
    shadowColor: 'gray',
    shadowOffset: { width: 0, height: 0.1 },
    shadowOpacity: 0.5,
    shadowRadius: 2,
  },
  btn: {
    alignSelf: 'flex-start',
    margin: 10,
    backgroundColor: '#131313',
    borderRadius: 50,
    padding: 10,
  },
  container1: {
    height: 220,
    width: '96%',
    marginTop: 0,
    marginHorizontal: 10,
    // borderRadius: 15,
    // elevation: 1,
    shadowColor: 'gray',
    shadowOffset: { width: 0, height: 0.1 },
    shadowOpacity: 0.5,
    shadowRadius: 2,
    alignItems: 'center',
  },
  image: {
    height: 190,
    width: '75%',
    marginVertical: 'auto',
    marginHorizontal: 'auto',
    resizeMode: 'contain',
    //   borderRadius: 10,
  },
  title: {
    color: '#FFF',
    position: 'absolute',
    marginHorizontal: 'auto',
    bottom: -30,
    fontSize: 22,
    width: '100%',
    textAlign: 'center',
    fontWeight: '600',
  },
  authors: {
    color: '#FFF',
    position: 'absolute',
    marginHorizontal: 'auto',
    bottom: -58,
    fontSize: 18,
    fontWeight: '500',
  },
  publisher: {
    color: '#FFF',
    position: 'absolute',
    marginHorizontal: 'auto',
    bottom: -90,
    fontSize: 18,
    fontWeight: '500',
  },
  date: {
    color: '#FFF',
    position: 'absolute',
    marginHorizontal: 'auto',
    bottom: -120,
    fontSize: 18,
    fontWeight: '500',
  },
  description: {
    fontSize: 16,
    color: '#FFF',
    marginHorizontal: 10,
    textAlign: 'start',
    marginBottom: 50,
  },
  descriptionContainer: {
    marginTop: 150,
    marginBottom: 10,
    backgroundColor: '#131313',
    padding: 10,
  },
  descTitle: {
    fontSize: 18,
    color: '#FFF',
    marginHorizontal: 10,
    marginVertical: 10,
    fontWeight: '700',
  },
});

export default BooksDetails;
