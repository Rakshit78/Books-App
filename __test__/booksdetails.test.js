import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';

import BooksDetails from '../components/BooksDetails';

describe('Render Books detail', () => {
  it('render booksdetails component without error', () => {
    const tree = render(<BooksDetails modal='hello' setModal={() => {}} />);
    expect(tree).toMatchSnapshot();
  });
});
