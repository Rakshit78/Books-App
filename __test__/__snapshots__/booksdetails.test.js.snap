// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Books detail render booksdetails component without error 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      [
        {
          "backgroundColor": "transparent",
        },
        {
          "flex": 1,
        },
      ]
    }
  >
    <ViewManagerAdapter_ExpoBlurView
      blurReductionFactor={4}
      experimentalBlurMethod="none"
      intensity={20}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      tint="default"
    />
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1A1A1A",
          "borderTopEndRadius": 15,
          "borderTopLeftRadius": 15,
          "flex": 1,
          "marginTop": "35%",
          "shadowColor": "gray",
          "shadowOffset": {
            "height": 0.1,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 2,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-start",
            "backgroundColor": "#131313",
            "borderRadius": 50,
            "margin": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "height": 220,
            "marginHorizontal": 10,
            "marginTop": 0,
            "shadowColor": "gray",
            "shadowOffset": {
              "height": 0.1,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 2,
            "width": "96%",
          }
        }
      >
        <Image
          resizeMode="cover"
          source={
            {
              "uri": undefined,
            }
          }
          style={
            {
              "height": 190,
              "marginHorizontal": "auto",
              "marginVertical": "auto",
              "resizeMode": "contain",
              "width": "75%",
            }
          }
        />
        <Text
          numberOfLines={2}
          style={
            {
              "bottom": -30,
              "color": "#FFF",
              "fontSize": 22,
              "fontWeight": "600",
              "marginHorizontal": "auto",
              "position": "absolute",
              "textAlign": "center",
              "width": "100%",
            }
          }
        />
        <Text
          numberOfLines={1}
          style={
            {
              "bottom": -58,
              "color": "#FFF",
              "fontSize": 18,
              "fontWeight": "500",
              "marginHorizontal": "auto",
              "position": "absolute",
            }
          }
        />
        <Text
          style={
            {
              "bottom": -90,
              "color": "#FFF",
              "fontSize": 18,
              "fontWeight": "500",
              "marginHorizontal": "auto",
              "position": "absolute",
            }
          }
        >
          Published Date: 
        </Text>
        <Text
          style={
            {
              "bottom": -120,
              "color": "#FFF",
              "fontSize": 18,
              "fontWeight": "500",
              "marginHorizontal": "auto",
              "position": "absolute",
            }
          }
        >
          Publisher:
           
          Not Available
        </Text>
      </View>
      <RCTScrollView
        style={
          {
            "backgroundColor": "#131313",
            "marginBottom": 30,
            "marginTop": 150,
            "padding": 10,
          }
        }
      >
        <View>
          <Text
            style={
              {
                "color": "#FFF",
                "fontSize": 18,
                "fontWeight": "700",
                "marginHorizontal": 10,
                "marginVertical": 10,
              }
            }
          >
            Description
          </Text>
          <Text
            style={
              {
                "color": "#FFF",
                "fontSize": 16,
                "marginBottom": 50,
                "marginHorizontal": 10,
                "textAlign": "start",
              }
            }
          />
        </View>
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
